contracts/PartyBid.sol

```diff
@@ -0,0 +1,701 @@
+            "PartyBid::bid: auction already finalized"
+        );
+        // get the minimum next bid for the auction
+        uint256 _bid = IMarketWrapper(marketWrapper).getMinimumBid(auctionId);
+        // ensure there is enough ETH to place the bid including PartyDAO fee
+        require(
+            _bid <= getMaximumBid(),
+            "PartyBid::bid: insufficient funds to bid"
+        );
+        // submit bid to Auction contract using delegatecall
+        (bool success, bytes memory returnData) =
+            marketWrapper.delegatecall(
+                abi.encodeWithSignature("bid(uint256,uint256)", auctionId, _bid)
+            );
+        require(
+            success,
+            string(
+                abi.encodePacked(
+                    "PartyBid::bid: place bid failed: ",
+                    returnData
+                )
+            )
+        );
+        // update highest bid submitted & emit success event
+        highestBid = _bid;
+        emit Bid(_bid);
+    }
+
+    // ======== External: Finalize =========
+
+    /**
+     * @notice Finalize the state of the auction
+     * @dev Emits a Finalized event upon success; callable by anyone
+     */
+    function finalize() external nonReentrant {
+        require(
+            partyStatus == PartyStatus.AUCTION_ACTIVE,
+            "PartyBid::finalize: auction not active"
+        );
+        // finalize auction if it hasn't already been done
+        if (!IMarketWrapper(marketWrapper).isFinalized(auctionId)) {
+            IMarketWrapper(marketWrapper).finalize(auctionId);
+        }
+        // after the auction has been finalized,
+        // if the NFT is owned by the PartyBid, then the PartyBid won the auction
+        address _owner = _getOwner();
+        partyStatus = _owner == address(this) ? PartyStatus.AUCTION_WON : PartyStatus.AUCTION_LOST;
+        uint256 _ethFee;
+        // if the auction was won,
+        if (partyStatus == PartyStatus.AUCTION_WON) {
+            // transfer ETH fee to PartyDAO
+            _ethFee = _getEthFee(highestBid);
+            _transferETHOrWETH(partyDAOMultisig, _ethFee);
+            // record total spent by auction + PartyDAO fees
+            totalSpent = highestBid + _ethFee;
+            // deploy fractionalized NFT vault
+            // and mint fractional ERC-20 tokens
+            _fractionalizeNFT();
+        }
+        // set the contract status & emit result
+        emit Finalized(partyStatus, totalSpent, _ethFee, totalContributedToParty);
+    }
+
+    // ======== External: Claim =========
+
+    /**
+     * @notice Claim the tokens and excess ETH owed
+     * to a single contributor after the auction has ended
+     * @dev Emits a Claimed event upon success
+     * callable by anyone (doesn't have to be the contributor)
+     * @param _contributor the address of the contributor
+     */
+    function claim(address _contributor) external nonReentrant {
```

Comment:
I am pretty confident that you can get rid of all these reentrancy guards with some slight reordering of operations, but that could easily change as features get added.

Reply:
yeah the thinking behind the re-entrancy guards was really just to be overly defensive and not have to think about the possibility of re-entrancy